### DISCUSSION TITLE
tB as VB-Script replacement

### DISCUSSION THREAD

**Discussion by uwbwsvd** [twinbasic/twinbasic#364] [Ideas]
Is it a crazy idea that tB can also replace VB-Script?

**Comment by WaynePhillipsEA** [comment]
Not a crazy idea at all.  I think it would be great if tB could open a VBS file and compile it to an EXE for example.  Along with integrated debugging capabilities of tB, it could be quite a useful tool.  Eventually of course :)

**Comment by Kr00l** [comment]
I think @uwbwsvd intention was to have a plain text file with a source, e.g. "MyScript.twinbasic" and run that source on the fly via an interpreter. But that would require a runtime like dependency in the system I guess.

**Comment by WaynePhillipsEA** [comment]
Ah ok, sorry, my misunderstanding.

So this would require a command line option on the compiler to read in the source file like a .twin file, compile it and execute Main().  Sounds simple enough.  Then you'd need to register the compiler command line for the .twinbasic extension appropriately in the registry for Windows to recognize it.  Again, not too difficult.

It's just a case of priorities, but we might be able to fit this in when we next touch something in the command-line area of the codebase, as I can see it would be useful.

**Reply by machthree** [reply]
Personal view here... 

While I like the idea of this in principle - I have written far too many VBScript scripts in my time - I would thi k carefully about the security implications of this - e.g. downloading from a browser, from an email etc. I suspect this would prove a major barrier to deployment of TB in any enterprise with without careful policy management tools. i.e. this is potentially a bigger job than it looks on the surface.

**Reply by WaynePhillipsEA** [reply]
We definitely wouldn't register the file extension in the Windows registry by default, so I don't see a major security concern.  It would be something to opt-in.  I see this as a useful feature for some, but most devs probably wouldn't use it as you can compile to tiny EXEs anyway.
